router/src/queue.rs

```diff
@@ -388,20 +415,57 @@ impl AdapterQueuesState {
             }
         }
 
+        // Add back cost for all offload adapters
+        for adapter in offload_adapters.iter() {
+            let queue = self.queue_map.get(adapter).unwrap().clone();
+            let cost = queue.cost().unwrap();
```

Comment:
Does this need a `none` check?

Reply:
I had one originally, but we never add to the active set until the cost non-none (see check below). So if this condition is violated, it should be a programming error.